tokio-net/src/signal/unix.rs

```diff
@@ -457,21 +460,4 @@ mod tests {
     fn with_timeout<F: Future>(future: F) -> impl Future<Output = F::Output> {
         Timeout::new(future, Duration::from_secs(1)).map(|result| result.expect("timed out"))
     }
-
-    #[tokio::test]
-    async fn ctrl_c() {
```

Comment:
IIRC there used to be problems with sending the SIGINT signal via the test harness, so this test lived in here and simulated a SIGINT delivery without actually exercising the OS hooks. Just curious if there was a reason to move it as an integration test?

Reply:
Unfortunately, I don't think it can live in a lib test as is. It couldn't find the reactor thread-local. I think this was due to referencing tokio which referenced back to the lib or something... 

If the test stays there, it would have to be rewritten to not use `#[tokio::test]`.

Reply:
Okay we can bring it back if it still causes problems. One last suggestion I have is to move the test to its own file, so that each test instance has its own fresh set of global variables and doesn't experience any interference with other tests